src/Symfony/Bundle/SecurityBundle/Resources/config/security.php

```diff
@@ -90,6 +90,8 @@
             ->args([service_locator([
                 'security.token_storage' => service('security.token_storage'),
                 'security.authorization_checker' => service('security.authorization_checker'),
+                'request_stack' => service('request_stack'),
+                'event_dispatcher' => service('event_dispatcher'),
```

Comment:
As far as I know the event dispatcher of the firewall responsible for the current request has to be used, not just the default event dispatcher. Otherwise logout listeners for the wrong firewalls may be triggered.

I am not entirely sure how this can be done at runtime though, probably all security event dispatchers would have to be wired in here.

Reply:
A service locator has been used. Thanks